### PR TITLE
Add testing strategy for Infrastructure Manager

### LINKED ISSUE #120: Define testing concept for Kyma Infrastructure Manager
**Description**

For our release management and to fulfil SAP product standards, we have to document how our testing strategy for the Infrastructure Manager looks like.

Some example links to such documentations are available here: https://wiki.one.int.sap/wiki/display/kyma/Testing+Strategy+-+Link+summary

The testing strategy should cover:

* How the different layers of the application are tests (e.g. unit-tests for structs  + packages, integration tests for inter-module objects, end2-end tests for API tests, performance tests for KPI measuring etc.)
* When tests tests are executed and what action gets triggered if a test fails
* How the test-coverage can be measured and what meaningful thresholds are and what happens if the threshold is not reached/exceeded.

AC:
- [ ] Create a testing strategy document for the Infrastructure Manager (proposal for the location: https://github.com/kyma-project/infrastructure-manager/tree/main/docs/contributor/...)
- [ ] Request feedback from @ameteiko about the testing strategy to ensure it's comprehensive 
- [ ] When the documentation is ready, update the link in this Wiki page: https://wiki.one.int.sap/wiki/display/kyma/Testing+Strategy+-+Link+summary

**Area**
Infrastructure Manager

**Reasons**

Mandatory part of the delivery process and required for a fast creation of Microdeliveries.

**Assignees**

@kyma-project/technical-writers

**Attachments**

### PR DESCRIPTION
**Pull request by grego952**
**Description**

Changes proposed in this pull request:

- Add testing strategy for Infrastructure Manager

**Related issue(s)**
Fixes #120

### CODE REVIEW

File: docs/contributor/testing-strategy.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- `Run unit tests / validate (pull_request)` - Executes basic create/update/delete functional tests of the reconciliation logic.
+- `Run vuln check / test (pull_request)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities.
+- `pull-infrastructure-manager-build` - builds the Docker image and pushes it to the registry.
+
+After the pull request is merged, the following CI/CD jobs are executed:
+
+ - `Run unit tests / validate (push)` - Executes basic create/update/delete functional tests of the reconciliation logic.
+ - `Run vuln check / test (push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities.
+ - `golangci-lint / lint (push)` - Is responsible for linting and static code analysis.
+ - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:16]: I'm a prow novice and I don't exactly know what is the difference between `pull-infrastructure-manager-build` and `main-infrastructure-manager-build`. Anyway I've added link to the responsible configuration file.
```suggestion
 - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```

**Review by Disper — CHANGES_REQUESTED** [review]

**Comment by Disper** [comment]
I'm not an expert in writing testing strategies but I see value in the CI inventory introduced in this PR.

File: docs/contributor/testing-strategy.md (hunk 1 above)
Comment (grego952) [reply docs/contributor/testing-strategy.md:16]: Maybe @ameteiko will be able to show the differences :)

File: docs/contributor/testing-strategy.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- `Run vuln check / test (pull_request)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-vuln-check.yaml).
+- `pull-infrastructure-manager-build` - builds the Docker image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
+
+After the pull request is merged, the following CI/CD jobs are executed:
+
+ - `Run unit tests / validate (push)` - Executes basic create/update/delete functional tests of the reconciliation logic.
+ - `Run vuln check / test (push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities.
+ - `golangci-lint / lint (push)` - Is responsible for linting and static code analysis.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:15]: I don't think that those are executed again after PR is merged.
```suggestion
```

Comment (Disper) [reply docs/contributor/testing-strategy.md:15]: I was mistaken.

File: docs/contributor/testing-strategy.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:16]:
```suggestion
Each pull request to the repository triggers the following CI/CD jobs:

- `golangci-lint / lint (pull_request/push)` - Is responsible for linting and static code analysis. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.golangci.yaml) and [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/golangci-lint.yaml).
- `PR Markdown Link Check / markdown-link-check (pull_request)` - Checks if there are no broken links in the pull request `.md` files. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/mlc.config.json).
- `Run unit tests / validate (pull_request/push)` - Executes basic create/update/delete functional tests of the reconciliation logic. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-tests.yaml).
- `Run vuln check / test (pull_request/push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-vuln-check.yaml).
- `pull-infrastructure-manager-build` - builds the Docker image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).

Additionally, after the pull request is merged, the following CI/CD job is executed:

 - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```

**Review by Disper — DISMISSED** [review]

File: docs/contributor/testing-strategy.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### Integration Testing
+1. The PO together with the team create an registry of implemented business features and define for each feature a suitable test scenario.
+2. Create a separate test suite for integration testing.
+3. Each test scenario is implemented in a separte test case. Use the Kubebuilder Test Framework and others to create test cases that interact with the Kubernetes cluster.  
+4. Test the interaction and integration of your custom resources, controllers, and other components with the Kubernetes API.
+5. Ensure test cases cover various aspects such as resource creation, updating, deletion, and handling of edge cases.
+6. Validate the correctness of event handling, reconciliation, and other control logic.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times and be applied for each PR. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:48]: Is this point still valid for integration testing?

File: docs/contributor/testing-strategy.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+## Introduction
+This testing strategy describes how the Framefrog team is testing the product `Kyma Infrastructure Manager`. It outlines the approach and methodologies to be used for testing all layers of this product to ensure the stability, reliability, and correctness.
+
+
+## Testing Methodology
+
+We investigate the product by separating it into layers:
+
+1. Business Logic
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:11]: What does the `business logic` mean? And how is it different than `Business Features`?

File: docs/contributor/testing-strategy.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+1. The PO together with the team create an registry of implemented business features and define for each feature a suitable test scenario.
+2. Create a separate test suite for integration testing.
+3. Each test scenario is implemented in a separte test case. Use the Kubebuilder Test Framework and others to create test cases that interact with the Kubernetes cluster.  
+4. Test the interaction and integration of your custom resources, controllers, and other components with the Kubernetes API.
+5. Ensure test cases cover various aspects such as resource creation, updating, deletion, and handling of edge cases.
+6. Validate the correctness of event handling, reconciliation, and other control logic.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times and be applied for each PR. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### End-to-End Testing
+1. Use Helm to create, deploy, and manage test clusters and environments that closely resemble the productive execution context.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:51]: I understand that Helm can make more sense over Kustomize as this is a tool of choice of SRE. But it would be good if we aligned on that in the team.

File: docs/contributor/testing-strategy.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+|Integration Testing||X||Only long running tests daily|
+|End-to-End Testing|||X|Daily|
+
+### Testing Tools and Frameworks
+Use the following tools and frameworks to implement the above-mentioned testing levels:
+
+- **GoTest**: For unit testing of Golang code.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
+- **Ginkgo and Gomega**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:67]: Duplicated line

Comment (Disper) [inline docs/contributor/testing-strategy.md:67]:
```suggestion
```

File: docs/contributor/testing-strategy.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+1. Identify critical functions, methods, and components that require testing.
+2. Write unit tests using GoUnit tests, Ginkgo and Gomega frameworks.
+3. Ensure tests cover various scenarios, edge cases, and possible failure scenarios. We try to verify business relevant logic with at least 65% code coverage.
+4. Test for both positive and negative inputs to validate the expected behavior.
+5. Mock external dependencies and use stubs or fakes to isolate the unit under test.
+6. Run unit tests periodically during development and before each PR to prevent regressions.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### Integration Testing
+1. The PO together with the team create an registry of implemented business features and define for each feature a suitable test scenario.
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:42]:
```suggestion
1. The PO and the team create a registry of implemented business features and define a suitable test scenario for each feature.
```

File: docs/contributor/testing-strategy.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+2. Create a separate test suite for integration testing.
+3. Each test scenario is implemented in a separte test case. Use the Kubebuilder Test Framework and others to create test cases that interact with the Kubernetes cluster.  
+4. Test the interaction and integration of your custom resources, controllers, and other components with the Kubernetes API.
+5. Ensure test cases cover various aspects such as resource creation, updating, deletion, and handling of edge cases.
+6. Validate the correctness of event handling, reconciliation, and other control logic.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times and be applied for each PR. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### End-to-End Testing
+1. Use Helm to create, deploy, and manage test clusters and environments that closely resemble the productive execution context.
+2. For shot living Kubernetes clusters, use K3d or other lightweight Kubernetes cluster providers.
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:52]:
```suggestion
2. For short-living Kubernetes clusters, use k3d or other lightweight Kubernetes cluster providers.
```

File: docs/contributor/testing-strategy.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+|End-to-End Testing|||X|Daily|
+
+### Testing Tools and Frameworks
+Use the following tools and frameworks to implement the above-mentioned testing levels:
+
+- **GoTest**: For unit testing of Golang code.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
+- **Ginkgo and Gomega**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
+- **K3d**: For creating short-living and lightweight Kubernetes clustes running within a Docker context.
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:68]:
```suggestion
- **k3d**: For creating short-living and lightweight Kubernetes clusters running within a Docker context.
```

File: docs/contributor/testing-strategy.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+- **Helm**: For deploying and managing test clusters and environments for end-to-end testing.
+- **K6:**: For performance and stress testing
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:70]:
```suggestion
- **k6:**: For performance and stress testing
```

File: docs/contributor/testing-strategy.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+|Framework|Unit Testing|Integration Testing|End-to-End Testing|
+|--|--|--|--|
+|GoTest| X |  |  |
+|Kubebuilder Test Framework| X | X | |
+|EnvTest| X | X |  |
+|Ginkgo| X | X |  |
+|Gomega| X | X |  |
+|K3d|  |  | X |
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:79]:
```suggestion
|k3d|  |  | X |
```

File: docs/contributor/testing-strategy.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+|Helm|  |  | X |
+|K6|  |  | X |
```
Comment (grego952) [inline docs/contributor/testing-strategy.md:81]:
```suggestion
|k6|  |  | X |
```

File: docs/contributor/testing-strategy.md (hunk 5 above)
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:11]:
```suggestion
1. Code
```

File: docs/contributor/testing-strategy.md (hunk 4 above)
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:48]:
```suggestion
7. Integration tests have to be executed fast to minimize roundtrip times and be applied for each PR. Long running tests should be excluded from frequently executed test runs and be triggered periodically (e.g. 4 times a day)
```

File: docs/contributor/testing-strategy.md (hunk 6 above)
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:51]:
```suggestion
1. Use a main stream Kubernetes management tool (e.g. [Helm](https://helm.sh/) or [Kustomize](https://kustomize.io/)) to create, deploy, and manage test clusters and environments that closely resemble the productive execution context.
```

**Review by tobiscr — CHANGES_REQUESTED** [review]

File: docs/contributor/testing-strategy.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+We investigate the product by separating it into layers:
+
+1. Code
+
+    Includes the technical frameworks (e.g. Kubebuilder) and custom Golang code.
+
+2. Business Features
+
+    Combines the business logic into feature which is consumed by our customers.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:17]:
```suggestion
    Combines the code into feature which is consumed by our customers.
```

File: docs/contributor/testing-strategy.md [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+3. Product Integration
+
+    Verifies how our product is integarted into the technical landscape, how it interacts with 3rd party systems and how it is accessible by customers or remote systems.
+ 
+For each layer, there is a dedicated testing approach used:
+
+1. **Unit Testing for Business Logic:** Writing and executing tests for individual functions, methods, and components to verify their behavior and correctness in isolation.
```
Comment (Disper) [inline docs/contributor/testing-strategy.md:25]:
```suggestion
1. **Unit Testing for Code:** Writing and executing tests for individual functions, methods, and components to verify their behavior and correctness in isolation.
```

**Review by Disper — APPROVED** [review]

File: docs/contributor/testing-strategy.md [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+2. **Integration Testing for Business Features:** Validating the integration and interaction between different components, modules, and services in the project.
+3. **End-to-End Testing:** Testing the application as a whole in a production-like environment, mimicking real-world scenarios to ensure the entire system functions correctly, is performing well and secure.
+
+
+## Testing Approach
+
+### Unit Testing
+1. Identify critical functions, methods, and components that require testing.
+2. Write unit tests using GoUnit tests, Ginkgo and Gomega frameworks.
+3. Ensure tests cover various scenarios, edge cases, and possible failure scenarios. We try to verify business relevant logic with at least 65% code coverage.
```
[the remark this answers is not in the document]
Comment (Disper) [reply docs/contributor/testing-strategy.md:35]: Unfortunately, the unit test coverage report does not yet exist.

File: docs/contributor/testing-strategy.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+4. Test for both positive and negative inputs to validate the expected behavior.
+5. Mock external dependencies and use stubs or fakes to isolate the unit under test.
+6. Run unit tests periodically during development and before each PR to prevent regressions.
+7. Unit tests have to be executed as fast as possible to minimize roundtrip times. Long running tests should be excluded from requently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### Integration Testing
+1. The PO and the team create a registry of implemented business features and define a suitable test scenario for each feature.
```
[the remark this answers is not in the document]
Comment (Disper) [reply docs/contributor/testing-strategy.md:42]: AFAIU @tobiscr plans to creates one but it's not yet there.

File: docs/contributor/testing-strategy.md [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+5. Ensure test cases cover various aspects such as resource creation, updating, deletion, and handling of edge cases.
+6. Validate the correctness of event handling, reconciliation, and other control logic.
+7. Integration tests have to be executed fast to minimize roundtrip times and be applied for each PR. Long-running tests should be excluded from frequently executed test runs and be triggered periodically (e.g. 4 times a day)
+
+### End-to-End Testing
+1. Use a mainstream Kubernetes management tool (e.g. [Helm](https://helm.sh/) or [Kustomize](https://kustomize.io/)) to create, deploy, and manage test clusters and environments that closely resemble the productive execution context.
+2. For short-living Kubernetes clusters, use k3d or other lightweight Kubernetes cluster providers.
+3. Run regularly, but at least once per release, a performance test that measures product KPIs to indicate KPI violations or performance differences between release candidates.
+
+|Testing Approach|Per Commit|Per PR|Per Release|In intervals|
```
[the remark this answers is not in the document]
Comment (Disper) [reply docs/contributor/testing-strategy.md:55]: Good point. This part document in general describes our target state, not the current one and we will first have to improve the e2e tests and create the release strategy document.

File: docs/contributor/testing-strategy.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/contributor/testing-strategy.md:42]: Yip, not yet ready unfortunately

File: docs/contributor/testing-strategy.md [hunk 19]
```diff
@@ -0,0 +1,94 @@
+# Testing Strategy for Infrastructure Manager
+
+## Introduction
+This testing strategy describes how the Framefrog team tests Kyma Infrastructure Manager. It outlines the approach and methodologies used for testing all layers of this product to ensure stability, reliability, and correctness.
```
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:4]:
```suggestion
This testing strategy describes how the Framefrog team is testing the product `Kyma Infrastructure Manager`. It outlines the approach and methodologies to be used for testing all layers of this product to ensure:

* stability

    by applying load and performance tests which verify the product resilience under load peaks.

* reliability

    running chaos tests and operational awareness workshops ensure the reliablitiy of the product in a productive context.

* functional correctness

    by documenting and tracing all product features, their acceptance criterias and the used end-2-end tests to verify a correct implementation.

*  maintainability

    by regularly measuring of code quality metrics like reusability, modularity and package qualities.
```

File: docs/contributor/testing-strategy.md [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+
+- **GoTest**: For unit testing of Golang code.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
+- **Ginkgo and Gomega**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **k3d**: For creating short-living and lightweight Kubernetes clusters running within a Docker context.
+- **Helm**: For deploying and managing test clusters and environments for end-to-end testing.
+- **k6:**: For performance and stress testing
+
+|Framework|Unit Testing|Integration Testing|End-to-End Testing|
+|--|--|--|--|
```
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:72]:
```suggestion
|--|--|--|--|
|Golangci-lint| X |  |  |
```

File: docs/contributor/testing-strategy.md [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+
+### Testing Tools and Frameworks
+Use the following tools and frameworks to implement the above-mentioned testing levels:
+
+- **GoTest**: For unit testing of Golang code.
+- **Kubebuilder Test Framework and EnvTest**: For creating and executing integration tests that interact with the Kubernetes API.
+- **Ginkgo and Gomega**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **k3d**: For creating short-living and lightweight Kubernetes clusters running within a Docker context.
+- **Helm**: For deploying and managing test clusters and environments for end-to-end testing.
+- **k6:**: For performance and stress testing
```
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:69]:
```suggestion
- **[Golangci-lint](https://github.com/golangci/golangci-lint)**: Golang code linting for better code quality.
- **[Go-critic](https://github.com/go-critic/go-critic)**: Another linter for measruing differnt code quality metrics.
- **GoTest**: For unit testing of Golang code.
- **Kubebuilder Framework and [EnvTest](https://book.kubebuilder.io/reference/envtest.html)**: For creating and executing integration tests that interact with the Kubernetes API.
- **[Ginkgo](https://github.com/onsi/ginkgo) and [Gomega](https://github.com/onsi/gomega)**: For writing and executing unit tests with a BDD-style syntax and assertions.
- **[K3d](https://k3d.io/)**: For creating short-living and lightweight Kubernetes clustes running within a Docker context.
- **[Helm](https://helm.sh/)**: For deploying and managing test clusters and environments for end-to-end testing.
- **[K6](https://k6.io/)**: For performance and stress testing
```

**Review by tobiscr — CHANGES_REQUESTED** [review]

**Review by tobiscr — DISMISSED** [review]

File: docs/contributor/testing-strategy.md [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+- **Kubebuilder Test Framework and [EnvTest](https://book.kubebuilder.io/reference/envtest.html)**: For creating and executing integration tests that interact with the Kubernetes API.
+- **[Ginkgo](https://github.com/onsi/ginkgo) and [Gomega](https://github.com/onsi/gomega)**: For writing and executing unit tests with a BDD-style syntax and assertions.
+- **[k3d](https://k3d.io/)**: For creating short-living and lightweight Kubernetes clusters running within a Docker context.
+- **[Helm](https://helm.sh/)**: For deploying and managing test clusters and environments for end-to-end testing.
+- **[k6](https://k6.io/):**: For performance and stress testing.
+
+|Framework|Unit Testing|Integration Testing|End-to-End Testing|
+|--|--|--|--|
+|Golangci-lint| X |  |  |
+|Go-critic| X |  |  |
```
Comment (tobiscr) [inline docs/contributor/testing-strategy.md:81]:
```suggestion
```

**Review by tobiscr — APPROVED** [review]

File: docs/contributor/testing-strategy.md [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+- `Run unit tests / validate (pull_request/push)` - Executes basic create/update/delete functional tests of the reconciliation logic. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-tests.yaml).
+- `Run vuln check / test (pull_request/push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities. It's configured [here](https://github.com/kyma-project/infrastructure-manager/blob/main/.github/workflows/run-vuln-check.yaml).
+- `pull-infrastructure-manager-build` - Triggered with each pull request. It builds the Docker image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).	
+- `main-infrastructure-manager-build` - Triggered after the merge. Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```
Comment (IwonaLanger) [inline docs/contributor/testing-strategy.md:103]: It is recommended to use more [descriptive text for links](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/06-links.md#links). How about: `For details, see the [configuration file](link)`?

**Review by tobiscr — DISMISSED** [review]

**Review by IwonaLanger — APPROVED** [review]
